src/components/Calculator.js

```diff
@@ -1,38 +1,75 @@
+import React, { useState } from 'react';
 import Display from './Display';
+import calculate from '../logic/calculate';
 
 export default function Calculator() {
+  const [value, setValue] = useState({ total: '0', next: null, operation: null });
+
+  function handleClick(e) {
+    setValue((value) => ({
+      ...value,
+      ...calculate(value, e.target.name),
+    }));
+  }
+
   return (
     <div className="calculator-div">
-      <Display />
-      <button type="button">AC</button>
-      <button type="button">+/-</button>
-      <button type="button">%</button>
-      <button type="button" className="orange">
+      <Display value={value} />
+      <button type="button" onClick={handleClick} name="AC">
+        AC
+      </button>
+      <button type="button" onClick={handleClick} name=" +/-">
+        +/-
+      </button>
+      <button type="button" onClick={handleClick} name="%">
+        %
+      </button>
+      <button type="button" className="orange" name="÷">
```

Comment:
- [Required] Good job implementing the calculator events, however, you forgot to register an event listener on the division button which makes it not function the way it's supposed to, kindly rectify this.